examples/run_demo_ursula_fleet.py

```diff
@@ -0,0 +1,73 @@
+"""
+This file is part of nucypher.
+nucypher is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+nucypher is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+You should have received a copy of the GNU General Public License
+along with nucypher.  If not, see <https://www.gnu.org/licenses/>.
+"""
+
+# WARNING This is not a mining script!
+# you will not perform any re-encryptions, and you will not get paid.
+# It might be (but might not be) useful for determining whether you have
+# the proper dependencies and configuration to run an actual mining node.
+import os
+from contextlib import suppress
+from functools import partial
+
+from twisted.internet import reactor
+
+from nucypher.characters.lawful import Ursula
+
+FLEET_POPULATION = 12
+DEMO_NODE_STARTING_PORT = 11500
+
+ursula_maker = partial(Ursula, rest_host='127.0.0.1',
+                       federated_only=True,
+                       domains=["finnegans/heartbeat/whatever"]
+                       )
+
+
+def spin_up_federated_ursulas(quantity: int = FLEET_POPULATION):
+    # Ports
+    starting_port = DEMO_NODE_STARTING_PORT
+    ports = list(map(str, range(starting_port, starting_port + quantity)))
+
+    ursulas = []
+
+    sage = ursula_maker(
+        rest_port=ports[0],
+        db_filepath="sage.db",
+    )
+
+    ursulas.append(sage)
+    for index, port in enumerate(ports[1:]):
+        u = ursula_maker(
+            rest_port=port,
+            seed_nodes=[sage.seed_node_metadata()],
+            start_learning_now=True,
+            db_filepath=f"{port}.db",
+        )
+        ursulas.append(u)
+    for u in ursulas:
+        deployer = u.get_deployer()
+        deployer.addServices()
+        deployer.catalogServers(deployer.hendrix)
+        deployer.start()
+        print(f"{u}: {deployer._listening_message()}")
+    try:
+        reactor.run()  # GO!
+    finally:
+        with suppress(FileNotFoundError):
+            os.remove("sage.db")
+        for u in ursulas[1:]:
+            with suppress(FileNotFoundError):
+                os.remove(f"{u.rest_interface.port}.db")
+
+if __name__ == "__main__":
+    spin_up_federated_ursulas()
```

Comment:
If the intention of this script is just for demos, 12 seems like overkill given that the demos do 2-of-3 thresholds. Perhaps we could reduce the default and allow a command-line argument to be specified to override the default quantity (whatever it ends up being). Thoughts?

Reply:
Yeah, I don't disagree.  The thing is: Alice and Bob now bail on trying to publish / find the `TreasureMap` if there are less than 8 nodes (I chose 8 arbitrarily).  If we *don't* do this, then it's possible for Alice to publish to too few nodes (as there may be too few which are `matching_nodes_among(nodes_she_knows_about_at_that_moment)`, leading to a broken Policy.

It also seems to me that nodes, in federated mode anyway, are super cheap; this script spins up 12 of them no problem, but can also do 100 without issue on my laptop.

Reply:
...but I do basically agree with you.  Part of this is that #2124 needs to be closed.  Then I'll have less anxiety about publishing to so few nodes.